sympy/tensor/toperators.py

```diff
@@ -74,7 +74,7 @@ def _contract_indices_for_derivative(cls, expr, variables):
                 variables_opposite_valence.append(i)
 
         args, indices, free, dum = TensMul._tensMul_contract_indices(
-            [expr] + variables_opposite_valence, replace_indices=True)
+            [S(expr)] + variables_opposite_valence, replace_indices=True)
```

Comment:
Why does this have to be sympified?
I wonder what happens after this skips sympify. If there is a problem, it should not be sympified inside private methods because it's expected to accept already sanitized types.

Reply:
If we don't sympify `expr`, then there is an error in this line:
```
File "/home/namannimmo/sympy/sympy/tensor/toperators.py", line 107, in _expand_partial_derivative
  if not args[0].free_symbols:
AttributeError: 'int' object has no attribute 'free_symbols'
```
So maybe that's the reason for converting it into a Sympy Integer.

Reply:
I think that `PartialDerivative` should sympify in `__new__`. Can you add the sympify logic there?
`_contract_indices_for_derivative` is only used inside there.

Reply:
I wasn't sure if the author wanted to be able to express `PartialDerivative(const, var)` as unevaluated so I didn't do the reduction to zero in `__new__`, but you are right that sympification can be done there.